File: TransitCalculator.java
import java.util.ArrayList;
import java.util.Arrays;


public class TransitCalculator {

    int daysUsing;
    int ridesTaken;



    String[] NYCTransitOptions = { "Pay-per-ride (single ride)", "7-day Unlimited Rides", "30-day Unlimited Rides" };

    double[] NYCTransitFairs = { 2.75, 33.00, 127.00 };

    double[] finalFairs;

    public TransitCalculator() {
        finalFairs = new double[3];
        
        
    }

    public double unlimited30Price(double daysRiding, double rides) {
        return ((Math.ceil(daysRiding % 30) * 127) / rides);

    }

    public double unlimited7Price(double daysRiding, double rides) {
        return ((Math.ceil(daysRiding % 7) * 33) / rides);

    }

    public double payPerRide(double daysRiding, double rides) {
        return (2.75 * rides);

    }

    public double[] getRidePrices(double daysRiding, double rides) {
        

        finalFairs[0] = payPerRide(daysRiding, rides);
        finalFairs[1] = unlimited7Price(daysRiding, rides);
        finalFairs[2] = unlimited30Price(daysRiding, rides);

        return finalFairs;

    }

  public String getBestFair(double daysRiding, double rides){
    
    getRidePrices(daysRiding, rides);
    
    double lowest = finalFairs[0];

    int lowestIndex = 0;
    
    System.out.println(lowest);
    System.out.println(Arrays.toString(finalFairs));
    for (int i = 0; i<finalFairs.length; i++) {
        double rate = finalFairs[i];
    if (rate < lowest){
        lowest = rate;
        lowestIndex = i;
      }
    }
    System.out.println(lowest);
    
    System.out.println(Arrays.toString(finalFairs));
    String lowestRate = NYCTransitOptions[lowestIndex];
    return "you should get the " + lowestRate + "option at $" + lowest + " per ride";
  }

    public static void main(String[] args) {
        TransitCalculator firstOne = new TransitCalculator();
        System.out.println(firstOne.getBestFair(2.00, 4.00));

    }
}

File: Transit_Calculator.py

import math
class TransitCalculator:


    def __init__(self, days, rides):
        self.days = days
        self.rides = rides
        self.final_fairs = []
        self.nyc = ["Pay-per-ride (single ride)", "7-day Unlimited Rides", "30-day Unlimited Rides"]
        self.nyc_fairs = [2.75, 33.00, 127.00]
    


    def unil_30_pri(self):
        return((math.ceil(self.days % 30) * 127) / self.rides)

    def unil_7_pri(self):
        return((math.ceil(self.days % 7) * 33) / self.rides)

    def pay_per_ride(self):
        return(2.75 * self.rides)

    def get_ride_prices(self):
        self.final_fairs.extend([self.pay_per_ride(),self.unil_7_pri(), self.unil_30_pri()])
        return self.final_fairs

    def get_Best_Fair(self):
        self.get_ride_prices()

        lowest = self.final_fairs[0]
        lowest_index = 0

        for i in range(len(self.final_fairs)-1):
            if self.final_fairs[i] < lowest:
                lowest = self.final_fairs[i]
            lowest_index +=1
        
        lowest_rate = self.nyc[lowest_index]

        return f"you should get the {lowest_rate} option at ${lowest} per ride"

            




Nyc = TransitCalculator(2, 4)
print(Nyc.get_Best_Fair())